articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,331 @@
+
+The general loop structure refers to the fundamental components of loops in all languages and usage. All the loop types mentioned above have the following characeristics:
+
+- *Counter Variable* - We also know this as the initializer. It is the starting point of the loop where it executes. 
+
+- *Condition Check* - This is the condition a loop has to start or execute. After it's met, it ends.
+
+- *Increment or Decrement Counter Variable*  - This refers to the number in which counter variable increases or decreases until the loop meets the condition.
+
+
+### Loops in Dart
+We should note that the syntax of Dart Loops is like the ones in Java Programming Language.
+
+There are three loops in Dart:
+
+1. #### For Loop
+
+For Loop is an example of [Definite Loops](#definite-loops).
+
+>Syntax of For Loop.
+
+```Dart
+void main(){
+
+   for (initializer; condition; increment/decrement){
+  
+   //put your code here;
+
+}
+
+}
+
+```
+
+>How the For Loop works.
+
+The flow of the For Loop: 
+
+**Initialize - Condition Check - Code Execute - Increment**.
+
+The counter variable initializes, then it checks the condition if it's true the code executes, then the counter variable increases depending on the value of increment. The increased counter variable initializes, and it re-checks the condition. If it's still true, the code executes and so on until the condition check is false or not met, then the loop ends.
+
+
+***Example:***
+
+```Dart
+
+void main(){
+for(var i = 0; i<4; i++){
+    print ("Hello");
+ 
+ }
+
+}
+
+/** Output : Hello
+             Hello
+             Hello
+             Hello **/
+
+```
+Here the value of the initializer of the counter variable will increase by one from 0 based on the increment specified. If the value of i becomes equal to four or more, the loop terminates/ends since the condition states that the value of i should be less than 4. The loop is to print hello if the condition is true, hence the output is hello written four times (0,1,2,3) < 4. 
+
+2. #### While Loop
+
+While Loop is an example of [Indefinite Loops](#indefinite-loops).
+
+>Syntax of While Loop
+```Dart
+
+void Main(){
+   // initialize counter variable;
+
+   while(condition){
+      
+      //code;
+      
+      increment/decrement;
+
+   }
+
+}
+
+```
+
+>How the While Loop works.
+
+The flow of the While Loop: 
+
+**Condition Check - Code Execute - Increment**.
+
+In the While Loop, we initialization of the counter variable outside the syntax. It begins with the condition check, if it's true, the code executes. The value of the counter variable is then increments or decrements, and the new counter variable goes through the condition check. It will continue until the condition check is false.
+
+***NOTE:*** 
+#### We must always have the increment or decrement part of the While Loop, failure to which we create a never-ending loop.
```

Comment:
If you want to highlight this, either use blockquote or you can make the text bolder.

Reply:
Also, it's `failure to which will create a never-ending loop.`